File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/faculties/FacultiesActivity.java
package ru.ulstu_team.ulstuschedule.ui.faculties;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.design.widget.NavigationView;
import android.support.design.widget.Snackbar;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.view.View;

import java.util.List;

import javax.inject.Inject;

import ru.ulstu_team.ulstuschedule.R;
import ru.ulstu_team.ulstuschedule.data.model.Faculty;
import ru.ulstu_team.ulstuschedule.databinding.ActivityFacultiesBinding;
import ru.ulstu_team.ulstuschedule.ui.base.BaseActivity;

public class FacultiesActivity extends BaseActivity
        implements NavigationView.OnNavigationItemSelectedListener, FacultiesMvpView {

    @Inject FacultiesAdapter mAdapter;
    @Inject FacultiesPresenter mPresenter;
    private ActivityFacultiesBinding b;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        b = DataBindingUtil.setContentView(this, R.layout.activity_faculties);
        getActivityComponent().inject(this);

        b.rvFaculties.setLayoutManager(new LinearLayoutManager(this));
        b.rvFaculties.setItemAnimator(new DefaultItemAnimator());
        b.rvFaculties.setHasFixedSize(true);

        mPresenter.attachView(this);
        mPresenter.loadFaculties();

        b.srlRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                mPresenter.loadFaculties();
                b.srlRefresh.setRefreshing(false);
            }
        });
    }

    @Override
    public void showFaculties(List<Faculty> faculties) {
        mAdapter.setFaculties(faculties);
        b.rvFaculties.setAdapter(mAdapter);
    }

    @Override
    public void showEmptyList() {

    }

    @Override
    public void showError() {
        Snackbar.make(b.drawerLayout, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        mPresenter.loadFaculties();
                    }
                }).show();
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/faculties/FacultiesActivity.kt
package ru.ulstu_team.ulstuschedule.ui.faculties

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.design.widget.Snackbar
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.Toolbar
import ru.ulstu_team.ulstuschedule.R
import ru.ulstu_team.ulstuschedule.data.model.Faculty
import ru.ulstu_team.ulstuschedule.databinding.ActivityFacultiesBinding
import ru.ulstu_team.ulstuschedule.ui.base.BaseActivity
import javax.inject.Inject

class FacultiesActivity : BaseActivity(),
        NavigationView.OnNavigationItemSelectedListener, FacultiesMvpView {

    @Inject lateinit var mAdapter: FacultiesAdapter
    @Inject lateinit var mPresenter: FacultiesPresenter
    private lateinit var b: ActivityFacultiesBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        b = DataBindingUtil.setContentView<ActivityFacultiesBinding>(this, R.layout.faculties_content)
        activityComponent.inject(this)

        val toolbar = findViewById(R.id.toolbar) as Toolbar
        toolbar.title = getString(R.string.faculties)
        setSupportActionBar(toolbar)

        b.rvFaculties.layoutManager = LinearLayoutManager(this)
        b.rvFaculties.itemAnimator = DefaultItemAnimator()
        b.rvFaculties.setHasFixedSize(true)

        mPresenter.attachView(this)
        mPresenter.loadFaculties()

        b.srlRefresh.setOnRefreshListener {
            mPresenter.loadFaculties()
        }
    }

    override fun showFaculties(faculties: List<Faculty>) {
        mAdapter.setFaculties(faculties)
        b.rvFaculties.adapter = mAdapter
        b.srlRefresh.isRefreshing = false
    }

    override fun showEmptyList() {
        b.srlRefresh.isRefreshing = false
    }

    override fun showError() {
        Snackbar.make(b.drawerLayout, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить") { mPresenter.loadFaculties() }
                .show()
        b.srlRefresh.isRefreshing = false
    }
}
